### ISSUE TITLE
[DEVEL] Failed to send Eth transaction & Error Getting balance.

### ISSUE THREAD

**Issue by Inbell1s** [smartcontractkit/chainlink#5888]
**Description**
After adding new chains and their respective nodes, occasionally the node doesn't select the right key pair to execute the transaction.

**Your Environment**
ROOT=/chainlink
LOG_LEVEL=debug
CHAINLINK_TLS_PORT=8559
SECURE_COOKIES=true
ALLOW_ORIGINS=*
DATABASE_URL=
DATABASE_TIMEOUT=0
TLS_CERT_PATH=/chainlink/.tls/server.crt
TLS_KEY_PATH=/chainlink/.tls/server.key

Added evmChainID: 4, 80001,42,421611 (first -> last)

**Basic Information**
Unable to send a response on chain:

```
[ERROR] can't create job run fluxmonitorv2/flux_monitor.go:995 absoluteThreshold=0 answer=50484123098 contract=REDACTED err=failed to send Eth transaction: Skipped Flux Monitor submission: cannot send transaction on chain ID 421611; eth key with address 0xBe90432feeBeF80B28e8a68Fb4BB8A5CedDA2Dc6 is pegged to chain ID 42 errVerbose=cannot send transaction on chain ID 421611; eth key with address 0xBe90432feeBeF80B28e8a68Fb4BB8A5CedDA2Dc6 is pegged to chain ID 42

github.com/smartcontractkit/chainlink/core/services/bulletprooftxmanager.(*BulletproofTxManager).checkStateExists
        /chainlink/core/services/bulletprooftxmanager/bulletprooftxmanager.go:338
github.com/smartcontractkit/chainlink/core/services/bulletprooftxmanager.(*BulletproofTxManager).CreateEthTransaction.func1
        /chainlink/core/services/bulletprooftxmanager/bulletprooftxmanager.go:303
github.com/smartcontractkit/chainlink/core/services/pg.SqlxTransaction
        /chainlink/core/services/pg/sqlx.go:48
github.com/smartcontractkit/chainlink/core/services/pg.Q.Transaction
        /chainlink/core/services/pg/q.go:153
github.com/smartcontractkit/chainlink/core/services/bulletprooftxmanager.(*BulletproofTxManager).CreateEthTransaction
        /chainlink/core/services/bulletprooftxmanager/bulletprooftxmanager.go:291
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*orm).CreateEthTransaction
        /chainlink/core/services/fluxmonitorv2/orm.go:119
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxAggregatorContractSubmitter).Submit
        /chainlink/core/services/fluxmonitorv2/contract_submitter.go:59
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).queueTransactionForBPTXM
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:1061
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).pollIfEligible.func3
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:983
github.com/smartcontractkit/chainlink/core/services/pg.sqlxTransactionQ
        /chainlink/core/services/pg/transaction.go:144
github.com/smartcontractkit/chainlink/core/services/pg.SqlxTransaction
        /chainlink/core/services/pg/sqlx.go:50
github.com/smartcontractkit/chainlink/core/services/pg.Q.Transaction
        /chainlink/core/services/pg/q.go:153
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).pollIfEligible
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:979
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).consume.func1
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:395
github.com/smartcontractkit/chainlink/core/recovery.WrapRecover
        /chainlink/core/recovery/recover.go:31
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).consume
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:394
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1581
Skipped Flux Monitor submission
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*orm).CreateEthTransaction
        /chainlink/core/services/fluxmonitorv2/orm.go:127
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxAggregatorContractSubmitter).Submit
        /chainlink/core/services/fluxmonitorv2/contract_submitter.go:59
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).queueTransactionForBPTXM
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:1061
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).pollIfEligible.func3
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:983
github.com/smartcontractkit/chainlink/core/services/pg.sqlxTransactionQ
        /chainlink/core/services/pg/transaction.go:144
github.com/smartcontractkit/chainlink/core/services/pg.SqlxTransaction
        /chainlink/core/services/pg/sqlx.go:50
github.com/smartcontractkit/chainlink/core/services/pg.Q.Transaction
        /chainlink/core/services/pg/q.go:153
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).pollIfEligible
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:979
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).consume.func1
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:395
github.com/smartcontractkit/chainlink/core/recovery.WrapRecover
        /chainlink/core/recovery/recover.go:31
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).consume
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:394
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1581
failed to send Eth transaction
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxAggregatorContractSubmitter).Submit
        /chainlink/core/services/fluxmonitorv2/contract_submitter.go:58
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).queueTransactionForBPTXM
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:1061
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).pollIfEligible.func3
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:983
github.com/smartcontractkit/chainlink/core/services/pg.sqlxTransactionQ
        /chainlink/core/services/pg/transaction.go:144
github.com/smartcontractkit/chainlink/core/services/pg.SqlxTransaction
        /chainlink/core/services/pg/sqlx.go:50
github.com/smartcontractkit/chainlink/core/services/pg.Q.Transaction
        /chainlink/core/services/pg/q.go:153
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).pollIfEligible
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:979
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).consume.func1
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:395
github.com/smartcontractkit/chainlink/core/recovery.WrapRecover
        /chainlink/core/recovery/recover.go:31
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).consume
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:394
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1581 jobID=27 latestAnswer=74510084561 logger=FluxMonitor reportableRound=72570 stacktrace=github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).pollIfEligible
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:995
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).consume.func1
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:395
github.com/smartcontractkit/chainlink/core/recovery.WrapRecover
        /chainlink/core/recovery/recover.go:31
github.com/smartcontractkit/chainlink/core/services/fluxmonitorv2.(*FluxMonitor).consume
        /chainlink/core/services/fluxmonitorv2/flux_monitor.go:394 threshold=1
```

**Steps to Reproduce**
Add multiple chains and nodes.
Set-up flux monitor jobs.

EDIT:
Also failing to retrieve the right key pair for the balance monitor.
2022-01-30T13:42:22.972Z [ERROR] BalanceMonitor: error getting balance for key 0xKEY

**Comment by Jemma1445** [comment]
We are really sorry for the inconvenience, kindly get back with your issue , if you don't mind we help you out

**Comment by andrejrakic** [comment]
@Inbell1s, thanks for raising this issue! Keep in mind that this feature is still in the Alpha phase of testing.

**Comment by andrejrakic** [comment]
@Inbell1s, this is fixed in `v1.3.0`. Closing this issue.
